### PR TITLE
Don't eagerly build type for External Type in Global symbols

### LINKED ISSUE #1212: Hang in Intellisense due to FormulaType.Build() eagerly expand types
FormulaType.Build eagerly expands the dataverse types  (see #371).
FormulaType.Build  should be a nop, lightweigt wrapper. But this expansion makes it a very heavy operation when using dataverse types. 

However, this causes things to blow up during intellisense as soon as we have a dataverse type for rowscope. Intellisense goes down each field type in rowscope, and Formula, 
with callstacks like this:
 	FxPlaygroundCore.ServiceClientWrapper.Log<Microsoft.Xrm.Sdk.OrganizationResponse>(System.Func<Microsoft.Xrm.Sdk.OrganizationResponse> func, string callerName, string param) Line 59	C#
>	FxPlaygroundCore.ServiceClientWrapper.Execute(Microsoft.Xrm.Sdk.OrganizationRequest request) Line 89	C#
 	Microsoft.PowerFx.Dataverse.DataverseExtensions.Execute.AnonymousMethod__0()	Unknown
 	Microsoft.PowerFx.Dataverse.DataverseExtensions.DataverseCall<Microsoft.Xrm.Sdk.Messages.RetrieveEntityResponse>(System.Func<Microsoft.Xrm.Sdk.Messages.RetrieveEntityResponse> call, string operationDescription)	Unknown
 	Microsoft.PowerFx.Dataverse.DataverseExtensions.Execute<Microsoft.Xrm.Sdk.Messages.RetrieveEntityRequest, Microsoft.Xrm.Sdk.Messages.RetrieveEntityResponse, Microsoft.Xrm.Sdk.Metadata.EntityMetadata>(Microsoft.Xrm.Sdk.IOrganizationService svcClient, Microsoft.Xrm.Sdk.Messages.RetrieveEntityRequest request, System.Func<Microsoft.Xrm.Sdk.Messages.RetrieveEntityResponse, Microsoft.Xrm.Sdk.Metadata.EntityMetadata> transform, out Microsoft.Xrm.Sdk.Metadata.EntityMetadata result)	Unknown

 	Microsoft.PowerFx.Dataverse.XrmMetadataProvider.TryGetEntityMetadata(string logicalName, out Microsoft.Xrm.Sdk.Metadata.EntityMetadata entityMetadata)	Unknown
 	Microsoft.PowerFx.Dataverse.CdsEntityMetadataProvider.TryGetXrmEntityMetadata(string logicalName, out Microsoft.Xrm.Sdk.Metadata.EntityMetadata xrmEntity)	Unknown
 	Microsoft.PowerFx.Dataverse.CdsEntityMetadataProvider.TryGetDataSource(string logicalName, string variableName, out Microsoft.PowerFx.Dataverse.DataverseDataSourceInfo dataSource)	Unknown
 	Microsoft.PowerFx.Dataverse.CdsEntityMetadataProvider.Microsoft.PowerFx.Core.Entities.Delegation.IExternalDataEntityMetadataProvider.TryGetEntityMetadata(string expandInfoIdentity, out Microsoft.PowerFx.Core.Functions.Delegation.IDataEntityMetadata entityMetadata)	Unknown
 	Microsoft.PowerFx.Core.Types.DType.TryGetEntityDelegationMetadata(out Microsoft.PowerFx.Core.Functions.Delegation.IDelegationMetadata metadata) Line 1158	C#
 	Microsoft.PowerFx.Types.FormulaType.GetExpandedEntityType(Microsoft.PowerFx.Core.Types.DType expandEntityType, string relatedEntityPath) Line 93	C#

*** We're expanding it eagerly with Build. 

 	Microsoft.PowerFx.Types.FormulaType.Build(Microsoft.PowerFx.Core.Types.DType type) Line 132	C#
 	Microsoft.PowerFx.Types.AggregateType.TryGetFieldType(string name, out Microsoft.PowerFx.Types.FormulaType type) Line 57	C#
 	Microsoft.PowerFx.Types.AggregateType.GetFieldType(string fieldName) Line 44	C#
 	Microsoft.PowerFx.Types.AggregateType.GetFieldTypes.AnonymousMethod__13_0(string field) Line 120	C#
 	System.Linq.Enumerable.WhereSelectEnumerableIterator<Microsoft.PowerFx.Core.Types.TypedName, System.__Canon>.MoveNext()	Unknown
 	Microsoft.PowerFx.SymbolTableOverRecordType.Microsoft.PowerFx.Core.Binding.IGlobalSymbolNameResolver.GlobalSymbols.get() Line 59	C#
 	
	Microsoft.PowerFx.ComposedReadOnlySymbolTable.GlobalSymbols.get() Line 96	C#
 	Microsoft.PowerFx.ComposedReadOnlySymbolTable.GlobalSymbols.get() Line 96	C#
 	Microsoft.PowerFx.Intellisense.IntellisenseHelper.AddSuggestionsForGlobals(Microsoft.PowerFx.Intellisense.IntellisenseData.IntellisenseData intellisenseData) Line 736	C#
 	Microsoft.PowerFx.Intellisense.IntellisenseHelper.AddSuggestionsForValuePossibilities(Microsoft.PowerFx.Intellisense.IntellisenseData.IntellisenseData intellisenseData, Microsoft.PowerFx.Syntax.TexlNode node) Line 789	C#
 	Microsoft.PowerFx.Intellisense.Intellisense.FirstNameNodeSuggestionHandler.TryAddSuggestionsForNodeKind(Microsoft.PowerFx.Intellisense.IntellisenseData.IntellisenseData intellisenseData) Line 69	C#
 	Microsoft.PowerFx.Intellisense.Intellisense.NodeKindSuggestionHandler.Run(Microsoft.PowerFx.Intellisense.IntellisenseData.IntellisenseData intellisenseData) Line 34	C#
 	Microsoft.PowerFx.Intellisense.Intellisense.Suggest(Microsoft.PowerFx.Intellisense.IntellisenseContext context, Microsoft.PowerFx.Core.Binding.TexlBinding binding, Microsoft.PowerFx.Syntax.Formula formula) Line 53	C#
 	Microsoft.PowerFx.Engine.Suggest(Microsoft.PowerFx.CheckResult checkResult, int cursorPosition) Line 355	C#

### PR DESCRIPTION
**Pull request by jas-valgotar**
Fixes #1212

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableOverRecordType.cs [hunk 1]
```diff
@@ -55,18 +56,36 @@ public SymbolTableOverRecordType(RecordType type, ReadOnlySymbolTable parent = n
         IEnumerable<KeyValuePair<string, NameLookupInfo>> IGlobalSymbolNameResolver.GlobalSymbols
         {
             get
-            {
-                foreach (var kv in _type.GetFieldTypes())
-                {
-                    yield return new KeyValuePair<string, NameLookupInfo>(kv.Name, Create(kv.Name, kv.Type));
+            {
+                if (_type._type.AssociatedDataSources.Any())
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableOverRecordType.cs:60]: Can you add a comment about what this is doing?
It's ok to also link to the bug 1212 in the comments here for future followup.

**Review by MikeStall — APPROVED** [review]

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**